source/extensions/common/tap/tap_config_base.h

```diff
@@ -151,6 +156,74 @@ class FilePerTapSink : public Sink {
   const envoy::config::tap::v3::FilePerTapSink config_;
 };
 
+// UDP sink definition
+class UdpTapSink : public Sink {
+public:
+  UdpTapSink(const envoy::extensions::tap_sinks::udp_sink::v3::UdpSink& config);
+  // below one is only for UT
+  UdpTapSink(Network::UdpPacketWriterPtr&& utUdpPacketWriter)
+      : udp_packet_writer_(std::move(utUdpPacketWriter)) {
+    // Construct a fake udp_server_address_ to let UT pass
```

Comment:
What is UT?
Also this seems somewhat hacky. Can you provide more details why this is needed?

Reply:
comment added